packages/cms-lib/__tests__/http.js

```diff
@@ -9,6 +11,26 @@ jest.mock('request-promise-native', () => ({
   get: jest.fn().mockReturnValue(Promise.resolve()),
   post: jest.fn().mockReturnValue(Promise.resolve()),
 }));
+
+jest.mock('request', () => ({
+  get: jest.fn(({ uri }) => {
+    const isValidPath = path => {
+      if (path === 'some/endpoint/path') {
+        return true;
+      } else {
+        return false;
+      }
+    };
+    return {
+      on: jest.fn((val, cb) => {
+        if (val === 'response') {
+          cb({ statusCode: isValidPath(uri) ? 200 : 404 });
+        }
```

Comment:
Instead of the logic for faking a 404 response being at the level of the mock definition like this, I think it is a bit nicer if the mocked response is closer to the test. This makes it easier to see what is being tested and how.